### PR TITLE
Added method CreateVolumeSnapshotClassesWithParameters

### PR DESCRIPTION
**Pull request by ak-px**
**What this PR does / why we need it**:
Added method CreateVolumeSnapshotClassesWithParameters , This is required for creating invalid volumesnapshotclass for failing csi backups for px-backup tests.

**Which issue(s) this PR fixes** (optional)
Closes #PB-7192
#PB-7190

**Special notes for your reviewer**:
Did a dry run along withhttps://github.com/portworx/torpedo/pull/2588   (ignore the failure, as the complete partial backup methods are not ready.)

https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/byoc/job/px-backup-on-demand-system-test-byoc/5375/
https://aetos.pwx.purestorage.com/resultSet/testSetID/665862

### CODE REVIEW

File: drivers/scheduler/dcos/dcos.go [hunk 1]
```diff
@@ -1000,6 +1000,14 @@ func (d *dcos) CreateVolumeSnapshotClasses(snapClassName string, provisioner str
 	}
 }
 
+func (d *dcos) CreateVolumeSnapshotClassesWithParameters(snapClassName string, provisioner string, isDefault bool, deletePolicy string, parameters map[string]string) (*volsnapv1.VolumeSnapshotClass, error) {
```
Comment (kshithijiyer-px) [inline drivers/scheduler/dcos/dcos.go:1003]: Add docstring

**Review by kshithijiyer-px — APPROVED** [review]
Minor comment else looks good to me

File: drivers/scheduler/dcos/dcos.go (hunk 1 above)
Comment (ak-px) [reply drivers/scheduler/dcos/dcos.go:1004]: Done

**Review by sn-px — APPROVED** [review]
LGTM

**Review by sabrarhussaini — APPROVED** [review]
LGTM

**Review by mkoppal-px — APPROVED** [review]
LGTM

**Pull request merged**